src/App.jsx

```diff
@@ -22,24 +26,81 @@ function App() {
     setDarkTheme(!darkTheme);
   };
 
+  const [accessToken, setAccessToken] = useState(undefined);
+  const [profile, setProfile] = useState(undefined);
+  const [loading, setLoading] = useState(true);
+
+  const { getAccessTokenSilently, user } = useAuth0();
+
+  React.useEffect(() => {
+    const getAccessToken = async () => {
+      try {
+        const tempToken = await getAccessTokenSilently({
+          audience: import.meta.env.VITE_AUDIENCE,
+          scopes: ["read:profile"],
+        });
+        setAccessToken(tempToken);
+      } catch (e) {
+        setLoading(false);
+      }
+    };
+    getAccessToken();
+  }, []);
+
+  // fetch profile
+
+  useEffect(() => {
+    const getProfile = async () => {
+      console.log("object");
+      const profileResponse = await FetchProfileEmail(user.email, accessToken);
+      setProfile(profileResponse);
+    };
+
+    if (accessToken) {
+      getProfile();
+    }
+  }, [accessToken]);
+  console.log(profile);
   return (
     <BrowserRouter>
       <Switch>
         <Route path="/502" exact component={BadGatewayPage} />
-        <Route path="/" exact component={LandingPage} />
+        <Route
+          path="/"
+          exact
+          component={LandingPage}
+          accessToken={accessToken}
+          loading={loading}
+        />
         <Route path="/logout" exact component={LogoutPage} />
         <ProtectedRoute path="/profile" component={Profile} />
         <ProtectedRoute
           path="/dashboard"
           component={() => (
-            <HomePage themeChange={themeChange} darkTheme={darkTheme} />
+            <HomePage
+              themeChange={themeChange}
+              darkTheme={darkTheme}
+              accessToken={accessToken}
+              setAccessToken={setAccessToken}
+              profile={profile}
+            />
           )}
         />
         <ProtectedRoute
           path="/transactionshistory"
-          component={TransactionsHistoryPage}
+          component={() => (
+            <TransactionsHistoryPage
+              accessToken={accessToken}
+              setLoading={setLoading}
+              profile={profile}
+            />
+          )}
+        />
+        <ProtectedRoute
```

Comment:
Careful of the inconsistency between how your routes are handling components - it may confused you at some point. Maybe stick to just one pattern, even if it's a little more complex looking